dace/libraries/blas/nodes/gemm.py

```diff
@@ -0,0 +1,276 @@
+
+        if c_inputs is None or node.beta == 0:
+            mul_out, mul_out_array = "_y", array_y
+            output_nodes = None
+        else:
+            mul_out, mul_out_array = tmp, array_tmp = sdfg.add_temp_transient(
+                shape_y, dtype_y, storage=storage)
+
+            access_tmp = state.add_read(tmp)
+            output_nodes = {mul_out: access_tmp}
+
+        # multiplication map
+        state.add_mapped_tasklet(
+            "_MatMult_",
+            {"__i%d" % i: "0:%s" % s
+             for i, s in enumerate([M, N, K])}, {
+                 "__a":
+                 dace.Memlet.simple(
+                     "_a", "__i2, __i0" if node.transA else "__i0, __i2"),
+                 "__b":
+                 dace.Memlet.simple(
+                     "_b", "__i1, __i2" if node.transB else "__i2, __i1")
+             },
+            mul_program, {
+                "__out":
+                dace.Memlet.simple(mul_out,
+                                   "__i0, __i1",
+                                   wcr_str="lambda x, y: x + y",
+                                   wcr_identity=0)
+            },
+            external_edges=True,
+            output_nodes=output_nodes)
+
+        if c_inputs is not None and node.beta != 0:
+            add_program = "__y = ({} * __c) + __tmp".format(_cast_to_dtype_str(node.beta, dtype_a))
+
+            # manually broadcasting C to [M, N]
+            if shape_c == [M, N]:
+                memlet_idx = '__i0, __i1'
+            elif shape_c == [1, N]:
+                memlet_idx = '0, __i1'
+            elif shape_c == [M, 1]:
+                memlet_idx = '__i0, 0'
+            elif shape_c == [
+                    N,
+            ]:
+                memlet_idx = '__i1'
+            else:
+                raise ValueError(
+                    "Could not broadcast input _c to ({}, {})".format(M, N))
+
+            # addition map
+            state.add_mapped_tasklet(
+                "_Add_",
+                {"__i%d" % i: "0:%s" % s
+                 for i, s in enumerate([M, N])}, {
+                     "__c": dace.Memlet.simple("_c", memlet_idx),
+                     "__tmp": dace.Memlet.simple(mul_out, "__i0, __i1"),
+                 },
+                add_program, {"__y": dace.Memlet.simple("_y", "__i0, __i1")},
+                external_edges=True,
+                input_nodes={mul_out: access_tmp})
+
+            sdfg.parent = parent_sdfg
+        sdfg.parent_sdfg = parent_sdfg
+
+        return sdfg
+
+    @staticmethod
+    def expansion(node, state, sdfg):
+        node.validate(sdfg, state)
+        if node.dtype is None:
+            raise ValueError("Data type must be set to expand " + str(node) +
+                             ".")
+        return ExpandGemmPure.make_sdfg(node, state, sdfg)
+
+
+@dace.library.node
+class Gemm(dace.graph.nodes.LibraryNode):
+    """Executes alpha * (A @ B) + beta * C. Broadcasting C is NOT supported.
```

Comment:
Is it really not supported? Some cases are, maybe which ones should be given here.

Reply:
Yup, that was an old docstring before I decided to add broadcasting to the library node